### DISCUSSION TITLE
How do I use a custom API key to shuffle channels with too many uploads?

### DISCUSSION THREAD

**Discussion by Voltflip** [NikkelM/Random-YouTube-Video#323] [Q&A] [answered]
This is what I'm getting 

"The channel you are shuffling from has too many uploads for the amount of API requests you can make. To protect the userbase, each user has a limited amount of requests they can make per day. To shuffle from channels with more uploads, please use a custom API key."

I've dug through the extension settings on Firefox and haven't been able to find an option for entering an API key. Is this feature available yet?

**Comment by NikkelM** [comment, marked as answer]
Hi @Voltflip, you can add a custom API key by expanding the `Advanced Settings` menu and enabling the `Use custom API key` option, this also works on Firefox:
![image](https://github.com/user-attachments/assets/99e752c4-a0e4-4a9e-b08a-c4a0e630fc03)
To get an API key, you should follow the official guidance provided by Google [here](https://developers.google.com/youtube/v3/getting-started).
The API key will be saved locally on your device and not shared with the extension, and you can disable the option or delete the key whenever you feel like it, which will revert you to the default daily usage limits.
